### PR TITLE
Add context menus to overlay panels/cards

### PR DESCRIPTION
**Pull request by Joehuu**
Only basic "view" option for now, for UX consistency like in song select highlighted menu items that are the same as left clicking them.

### CODE REVIEW

File: osu.Game/Overlays/News/NewsCard.cs [hunk 1]
```diff
@@ -161,5 +163,10 @@ private void load(OverlayColourProvider colourProvider)
 
             DateTimeOffset IHasCustomTooltip<DateTimeOffset>.TooltipContent => date;
         }
+
+        public MenuItem[] ContextMenuItems => new MenuItem[]
+        {
+            new OsuMenuItem("View news in browser", MenuItemType.Highlighted, Action),
```
Comment (Joehuu) [inline osu.Game/Overlays/News/NewsCard.cs:169]: Not localised since "in browser" is temporary, and removing it will cause confusion.

**Comment by bastianpedersen** [comment]
Tests?

File: osu.Game/Overlays/News/NewsCard.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Overlays/News/NewsCard.cs:169]: I don't know how temporary it is. News posts heavily use inline HTML, especially for things like audio or YouTube embeds, and I'm not sure we're ever supporting that.

Probably fine for now either way, but just food for thought I guess.

File: osu.Game/Users/UserPanel.cs [hunk 2]
```diff
@@ -41,14 +43,23 @@ protected UserPanel(APIUser user)
             User = user;
         }
 
-        [Resolved(canBeNull: true)]
-        private UserProfileOverlay profileOverlay { get; set; }
+        [Resolved]
+        private UserProfileOverlay? profileOverlay { get; set; }
```
[the remark this answers is not in the document]
Comment (bdach) [reply osu.Game/Users/UserPanel.cs:47]: For the record I've gone and checked android, it works fine. Not sure whether that should be ascribed to .NET 6 or to sourcegen'd DI, but it does seem to work.

File: osu.Game/Overlays/OnlineOverlay.cs [hunk 3]
```diff
@@ -46,10 +47,15 @@ protected OnlineOverlay(OverlayColourScheme colourScheme, bool requiresSignIn =
                         Children = new Drawable[]
                         {
                             Header.With(h => h.Depth = float.MinValue),
-                            content = new PopoverContainer
+                            new OsuContextMenuContainer
```
[the remark this answers is not in the document]
Comment (Joehuu) [reply osu.Game/Overlays/OnlineOverlay.cs:50]: There's another one in user profile overlay:
https://github.com/ppy/osu/blob/bcaff3785190dab1ec327abe06e2817b9be9d23d/osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs#L74

but it inherits `FullscreenOverlay` instead and should be refactored in another PR.

File: osu.Game/Users/UserPanel.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (frenzibyte) [reply osu.Game/Users/UserPanel.cs:47]: iOS works fine as well, fwiw.

**Comment by bdach** [comment]
Technically speaking the code looks fine, but I do find myself questioning if we really want all of these. To me it feels kind of useless to be adding single-menu context items.

Like, the beatmap and user cards already change visual state when hovered, do they really need an extra single-item context menu?

image
image

The news one is arguably even worse, because news panels also have tooltips, so you can get something like this which looks quite ridiculous IMO:

image

**Comment by Joehuu** [comment]
I mean, I can add more items (e.g. download, favorite) to the beatmap card if they feel useless atm. The user panel is less useless since I added the send message item when right clicking other users.

I can also remove the news one for now, since that is temporary. I can also remove the news one for now (the temporary tooltip will suffice).

**Comment by bdach** [comment]
Let's just remove from news for now I guess. The rest doesn't bother me that much.

**Review by peppy — APPROVED** [review]

**Pull request merged**